src/java/org/apache/cassandra/repair/autorepair/AutoRepair.java

```diff
@@ -0,0 +1,457 @@
+                                {
+                                    logger.error("Auto-repair for type {} is disabled hence not running repair", repairType);
+                                    repairState.setRepairInProgress(false);
+                                    return;
+                                }
+
+                                if (config.getRepairByKeyspace(repairType))
+                                {
+                                    if (AutoRepairUtils.keyspaceMaxRepairTimeExceeded(repairType, tableStartTime, tablesToBeRepaired.size()))
+                                    {
+                                        skippedTokenRanges += totalSubRanges - totalProcessedSubRanges;
+                                        logger.info("Keyspace took too much time to repair hence skipping it {}",
+                                                    keyspaceName);
+                                        break;
+                                    }
+                                }
+                                else
+                                {
+                                    if (AutoRepairUtils.tableMaxRepairTimeExceeded(repairType, tableStartTime))
+                                    {
+                                        skippedTokenRanges += totalSubRanges - totalProcessedSubRanges;
+                                        logger.info("Table took too much time to repair hence skipping it {}.{}",
+                                                    keyspaceName, tableName);
+                                        break;
+                                    }
+                                }
+                                Token childStartToken = token.left;
+                                Token childEndToken = token.right;
+                                logger.debug("Current Token Left side {}, right side {}", childStartToken
+                                                                                          .toString(), childEndToken.toString());
+
+                                ranges.add(new Range<>(childStartToken, childEndToken));
+                                totalProcessedSubRanges++;
+                                if ((totalProcessedSubRanges % config.getRepairThreads(repairType) == 0) ||
+                                    (totalProcessedSubRanges == totalSubRanges))
+                                {
+                                    int retryCount = 0;
+                                    Future<?> f = null;
+                                    while (retryCount <= config.getRepairMaxRetries())
+                                    {
+                                        RepairRunnable task = repairState.getRepairRunnable(keyspaceName,
+                                                                                            config.getRepairByKeyspace(repairType) ? tablesToBeRepaired : ImmutableList.of(tableName),
```

Comment:
Since this is being evaluated for each table, won't this mean that for each Table in a keyspace, it will issue a full keyspace repair?

Reply:
Ah nevermind, I see there is a break at line 340 that breaks out after evaluating over the first iteration.